zbar/sqcode.c

```diff
@@ -371,7 +371,7 @@ found_start:;
 	    border_len = 1;
 	    top_border = malloc(sizeof(sq_point));
 	    if (!top_border)
-		return 1;
+		goto free_borders;
```

Comment:
I think there is no leak because the top border is the first one that would be allocated. Skipping free_borders is a matter of speed.